.github/workflows/pull-request.yaml

```diff
@@ -10,3 +10,56 @@ jobs:
     name: PR
     uses: canonical/observability/.github/workflows/charm-pull-request.yaml@main
     secrets: inherit
+
+  test-against-coordinator:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the canonical/tempo-worker-k8s-operator repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          path: "worker"
+          # leave other args blank: it's checking out THIS branch/THIS repo
+
+      - name: Checkout the canonical/tempo-coordinator-k8s-operator repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          repository: canonical/tempo-coordinator-k8s-operator
+          path: "coordinator"
+          ref: main
+
+      - name: Set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: pip install tox~=4.2
+
+      - name: Setup Charmcraft's pip cache
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/snap/charmcraft/common/cache/charmcraft/
+          key: charmcraft-cache-${{ github.job }}-${{ strategy.job-index }}-${{ github.run_id }}-${{ github.run_attempt }}
+          restore-keys: charmcraft-cache
+
+      - name: Get IP range
+        id: ip_range
+        run: |
+          echo "ip_range=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')/32" >> $GITHUB_OUTPUT
+
+      - name: Setup operator environment (k8s)
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: 3.4/stable
+          provider: microk8s
+          channel: 1.26-strict/stable
+          microk8s-group: snap_microk8s
+          microk8s-addons: "hostpath-storage dns metallb:${{ steps.ip_range.outputs.ip_range }}"
+          charmcraft-channel: "2.x/stable"
+
+      - name: Run the coordinator charm's integration tests
+        run: cd coordinator; tox -vve integration
+        env:
+          TEMPO_WORKER_CHARM_PATH: ../worker
```

Comment:
Currently, the coordinator won't pick **this** worker, it'll pick the one on edge?
Is it part of a later work?

Reply:
well the envvar [has to be updated to "TEMPO_CHARM" but then it should work](https://github.com/canonical/tempo-coordinator-k8s-operator/blob/main/tests/integration/conftest.py#L31C34-L31C45), no?

Reply:
I think that is for the coordinator. We should have sth similar for the worker charm (option to not deploy from edge) in the coordinator repo.

Reply:
tandem pr: https://github.com/canonical/tempo-coordinator-k8s-operator/pull/39